### PR TITLE
Added 'delete' functionality to education entries

### LINKED ISSUE #297: Profile: Allow deletion of education items
Make the existing UI functional. 

Shouldn't require any API changes.

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

closes #297 

#### What's this PR do?

this adds the ability to delete entries in the education tab (using `Array.prototype.slice`).

#### Where should the reviewer start?

Check out the changes in `EducationTab.js`!

#### How should this be manually tested?

Try to delete education items! Make sure that only the one you click on gets deleted! etc!

#### Screenshots (if appropriate)

delete

### CODE REVIEW

**Comment by pdpinch** [comment]
@aliceriot that screenshot isn't looking right. Something happened to the left margin?

**Comment by aliceriot** [comment]
not sure what you mean, here's the form on master:

master_educ

and on my branch:

my_branch

do you mean the different icon spacing? I switched from using `<Button>` to using `<IconButton>`, which is what we're currently using on the work history form.

**Comment by aliceriot** [comment]
( I can easily switch that back )

**Comment by pdpinch** [comment]
The screenshot on the PR cover letter looks really whacked out. The ones in [your comment]( https://github.com/mitodl/micromasters/pull/348#issuecomment-220103203) are what I was expecting. 

But, we eventually want to get to what's in the mockups (https://github.com/mitodl/micromasters/issues/331)

File: static/js/components/EducationTab.js [hunk 1]
```diff
@@ -103,26 +111,33 @@ class EducationTab extends ProfileTab {
...
           </Cell>
         </Grid>;
       }
-    });
+    };
+    return profile['education'].map(educationRow);
   }
 
-  openEditEducationForm(level, educationId){
+  openEditEducationForm = (level, educationId) => {
```
Comment (noisecapella) [inline static/js/components/EducationTab.js]: Does using the arrow function mean we don't need to bind this?

Comment (aliceriot) [inline static/js/components/EducationTab.js]: yeah, exactly. when we declare an arrow function in the body of a class declaration `this` refers to the object instance at runtime.

**Comment by noisecapella** [comment]
Not a showstopper, but there's a quirk due to the fact that we save the entire profile when you click 'Save' in the dialog box. So you could delete an education, edit another education, click Save and the delete is saved too. When you click Cancel the delete is also reverted along with the data in the education.

**Comment by noisecapella** [comment]
:+1:

**Pull request merged**